src/main/java/duke/command/Command.java

```diff
@@ -0,0 +1,198 @@
+     *                      exception occurred
+     */
+    public void execute(TaskList tasks) throws IOException, DukeException {
+        ArrayList<Task> tList = tasks.tasks;
+        if (commandName.equals("list")) {
+            System.out.println("     Here are the tasks in your list:");
+            int counter = 1;
+            for(Task s : tList) {
+                if (s != null) {
+                    System.out.println("     " + counter + "." + s.toString());
+                    counter++;
+                } else {
+                    break;
+                }
+            }
+        } else if (commandName.split(" ")[0].equals("done")) {
+            int index = Integer.parseInt(commandName.split(" ")[1]);
+            tList.get(index-1).markAsDone();
+            System.out.println("     Nice! I've marked this task as done:");
+            System.out.println("       " + tList.get(index-1).toString());
+            Task currTask = tList.get(index-1);
+            tasks.tasks = tList;
+            String old = null;
+            if (currTask instanceof ToDo) {
+                old = "T | 0 | " + currTask.name;
+            } else if (currTask instanceof Deadline) {
+                old = "D | 0 | " + currTask.name;
+            } else if (currTask instanceof Event) {
+                old = "E | 0 | " + currTask.name;
+            }
+            BufferedReader reader = new BufferedReader(new FileReader("data/duke.txt"));
+            String oldContent = "";
+            String line = reader.readLine();
+            while (line != null) {
+                oldContent = oldContent + line + System.lineSeparator();
+                line = reader.readLine();
+            }
+            String newContent = null;
+            if (currTask instanceof ToDo) {
+                newContent = oldContent.replace(old, "T | 1 | " + currTask.name);
+            } else if (currTask instanceof Deadline) {
+                newContent = oldContent.replace(old, "D | 1 | " + currTask.name);
+            } else if (currTask instanceof Event) {
+                newContent = oldContent.replace(old, "E | 1 | " + currTask.name);
+            }
+            FileWriter writer = new FileWriter("data/duke.txt");
+            writer.write(newContent);
+            reader.close();
+            writer.close();
+        } else if (commandName.split(" ")[0].equals("delete")) {
+            int index = Integer.parseInt(commandName.split(" ")[1]);
+            System.out.println("     Noted. I've removed this task:");
+            System.out.println("       " + tList.get(index-1).toString());
+            System.out.println("     Now you have " + (tList.size()-1) + " tasks in the list.");
+            tList.remove(index-1);
+            tasks.tasks = tList;
+            BufferedReader reader = new BufferedReader(new FileReader("data/duke.txt"));
+            String oldContent = "";
+            String line = reader.readLine();
+            int counter = 0;
```

Comment:
Maybe a name that clearly explains the functionality of this counter would be better? (i.e. what is this counter for or what is it counting?)